gtk4/src/expression.rs

```diff
@@ -146,6 +146,69 @@ impl Expression {
             ))
         }
     }
+
+    // rustdoc-stripper-ignore-next
+    /// Create a [`gtk::PropertyExpression`] that looks up for `property_name`
+    /// with self as parameter. This is useful in long chains of [`gtk::Expressions`].
+    ///
+    /// # Example
+    ///
+    /// Create an expression describing `list_item->item->metadata->number` and bind it to
+    /// button's label.
+    ///
+    /// ```no_run
+    /// # use gtk4 as gtk;
+    /// use gtk::prelude::*;
+    ///
+    /// let factory = gtk::SignalListItemFactory::new();
+    /// factory.connect_setup(|_, list_item| {
+    ///     let button = gtk::Button::new();
+    ///     list_item.set_child(Some(&button));
+    ///
+    ///     list_item.property_expression("item")
+    ///         .chain_property("metadata")
+    ///         .chain_property("number")
+    ///         .bind(&button, "label", Some(&button));
+    /// });
+    /// ```
+    pub fn chain_property(&self, property_name: &str) -> crate::PropertyExpression {
+        crate::PropertyExpression::new(self.value_type(), Some(self), property_name)
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// Create a [`gtk::ClosureExpression`] with self as a parameter. This is useful in long
+    /// chains of [`gtk::Expressions`].
+    ///
+    /// # Examples
+    ///
+    /// Creates an expression describing `list_item->item->number` with closure that formats the
+    /// number with a prefix and bind to button's label.
+    ///
+    /// ```no_run
+    /// # use gtk4 as gtk;
+    /// use gtk::prelude::*;
+    ///
+    /// let factory = gtk::SignalListItemFactory::new();
+    /// factory.connect_setup(|_, list_item| {
+    ///     let button = gtk::Button::new();
+    ///     list_item.set_child(Some(&button));
+    ///
+    ///     list_item.property_expression("item")
+    ///         .chain_property("number")
+    ///         .chain_closure(|args| {
+    ///             let number: i32 = args[1].get().unwrap();
+    ///             format!("Number: {}", number)
+    ///         })
+    ///         .bind(&button, "label", Some(&button));
+    /// });
+    /// ```
+    pub fn chain_closure<F, R>(&self, f: F) -> crate::ClosureExpression
+    where
+        F: Fn(&[glib::Value]) -> R + 'static,
+        R: glib::value::ValueType,
+    {
+        crate::ClosureExpression::new(&[self.clone()], f)
```

Comment:
I think this could now take in an owned self?

Reply:
Good idea

Reply:
hmm, this causes `cannot move out of dereference of PropertyExpression`

Reply:
Oh well, let's keep the clone then and add a FIXME comment